app/src/main/java/org/schabi/newpipe/settings/SettingMigrations.java

```diff
@@ -0,0 +1,126 @@
+package org.schabi.newpipe.settings;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.preference.PreferenceManager;
+
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.report.ErrorActivity;
+import org.schabi.newpipe.report.ErrorActivity.ErrorInfo;
+import org.schabi.newpipe.report.UserAction;
+
+public final class SettingMigrations {
```

Comment:
Not in this PR, but we may consider adding junit tests for migrations, using some kind of mocked shared preferences

Reply:
Completely agree with that.